wasmedge fails to build from source with GCC 15 on amd64

The package wasmedge with version 0.14.1-r4 fails to build from source with GCC 15 on amd64 with the following error:

```2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2189:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Object/ObjectFile.h:22:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Object/Binary.h:17:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Object/Error.h:16:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/Support/Error.h:305:24: warning: unused parameter 'V' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   305 |   void setChecked(bool V) {
2025/05/31 00:59:49 INFO       |                        ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2189:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/Object/ObjectFile.h:296:49: warning: unused parameter 'SectionName' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   296 |   mapReflectionSectionNameToEnumValue(StringRef SectionName) const {
2025/05/31 00:59:49 INFO       |                                                 ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/Object/ObjectFile.h:346:38: warning: unused parameter 'TheTriple' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   346 |   virtual void setARMSubArch(Triple &TheTriple) const { }
2025/05/31 00:59:49 INFO       |                                      ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:20:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/ADT/SetVector.h:24:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/ADT/DenseSet.h:17:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/DenseMap.h:608:45: warning: unused parameter 'Key' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   608 |   BucketT *InsertIntoBucketImpl(const KeyT &Key, const LookupKeyT &Lookup,
2025/05/31 00:59:49 INFO       |                                             ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:21:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/BasicBlock.h:19:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/ilist.h:58:34: warning: unused parameter 'V' [-Wunused-parameter]
2025/05/31 00:59:49 INFO    58 |   static void deleteNode(NodeTy *V) {}
2025/05/31 00:59:49 INFO       |                                  ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:21:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/BasicBlock.h:23:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/DebugProgramInstruction.h:55:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/Instruction.h:18:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/Bitfields.h:125:32: warning: unused parameter 'UserMaxValue' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   125 |   static T pack(T UserValue, T UserMaxValue) {
2025/05/31 00:59:49 INFO       |                                ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/Bitfields.h:138:32: warning: unused parameter 'UserMaxValue' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   138 |   static T pack(T UserValue, T UserMaxValue) {
2025/05/31 00:59:49 INFO       |                                ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:22:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/Dominators.h:20:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/DepthFirstIterator.h:155:40: warning: unused parameter 'G' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   155 |   static df_iterator end(const GraphT &G) { return df_iterator(); }
2025/05/31 00:59:49 INFO       |                                        ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/ADT/DepthFirstIterator.h:161:40: warning: unused parameter 'G' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   161 |   static df_iterator end(const GraphT &G, SetType &S) { return df_iterator(S); }
2025/05/31 00:59:49 INFO       |                                        ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:22:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/Dominators.h:28:
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:441:38: warning: unused parameter 'IR' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   441 |   void verifyNotInvalidated(IRUnitT &IR, typename PassT::Result *Result) const {
2025/05/31 00:59:49 INFO       |                                      ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:441:66: warning: unused parameter 'Result' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   441 |   void verifyNotInvalidated(IRUnitT &IR, typename PassT::Result *Result) const {
2025/05/31 00:59:49 INFO       |                                                                  ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:624:23: warning: unused parameter 'IR' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   624 |   Result run(IRUnitT &IR, AnalysisManager<IRUnitT, ExtraArgTs...> &AM,
2025/05/31 00:59:49 INFO       |                       ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:624:68: warning: unused parameter 'AM' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   624 |   Result run(IRUnitT &IR, AnalysisManager<IRUnitT, ExtraArgTs...> &AM,
2025/05/31 00:59:49 INFO       |                                                                    ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:909:34: warning: unused parameter 'Arg' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   909 |   PreservedAnalyses run(IRUnitT &Arg, AnalysisManagerT &AM, ExtraArgTs &&...) {
2025/05/31 00:59:49 INFO       |                                  ^
2025/05/31 00:59:49 INFO /usr/lib/llvm-19/include/llvm/IR/PassManager.h:909:57: warning: unused parameter 'AM' [-Wunused-parameter]
2025/05/31 00:59:49 INFO   909 |   PreservedAnalyses run(IRUnitT &Arg, AnalysisManagerT &AM, ExtraArgTs &&...) {
2025/05/31 00:59:49 INFO       |                                                         ^
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/compiler.cpp:10:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/data.h:5:
2025/05/31 00:59:49 INFO In file included from /home/build/lib/llvm/llvm.h:2190:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Transforms/Utils/BasicBlockUtils.h:22:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/IR/Dominators.h:30:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Pass.h:351:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/PassSupport.h:29:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/PassRegistry.h:22:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Support/RWMutex.h:17:
2025/05/31 00:59:49 INFO In file included from /usr/lib/llvm-19/include/llvm/Support/Threading.h:21:
2025/05/31 00:59:49 INFO /usr/lib64/gcc/x86_64-pc-linux-gnu/15/../../../../include/c++/15/ciso646:46:4: error: "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros" [-Werror,-W#warnings]
2025/05/31 00:59:49 INFO    46 | #  warning "<ciso646> is deprecated in C++17, use <version> to detect implementation-specific macros"
2025/05/31 00:59:49 INFO       |    ^
2025/05/31 00:59:49 INFO 22 warnings and 1 error generated.
2025/05/31 01:00:15 WARN ninja: subcommands failed
2025/05/31 01:00:18 ERRO failed to build package: unable to run package wasmedge pipeline: unable to run pipeline: exit status 1
make[1]: *** [Makefile:157: packages/x86_64/wasmedge-0.14.1-r4.apk] Error 1
make[1]: Leaving directory '/home/sergio_durigan_chainguard_dev/os'
make: *** [Makefile:151: package/wasmedge] Error 2
```

--- a/sasl-xoauth2.yaml
+++ b/sasl-xoauth2.yaml
@@ -1,7 +1,7 @@
 package:
   name: sasl-xoauth2
   version: "0.27"
-  epoch: 0
+  epoch: 1
   description: "SASL plugin for XOAUTH2"
   copyright:
     - license: Apache-2.0
@@ -19,7 +19,6 @@ environment:
       - cmake-3
       - curl-dev
       - cyrus-sasl-dev
-      - gcc-14-default
       - jsoncpp-dev
       - openssl-dev
       - pkgconf

--- a/splunk-otel-collector.yaml
+++ b/splunk-otel-collector.yaml
@@ -1,7 +1,7 @@
 package:
   name: splunk-otel-collector
   version: "0.132.0"
-  epoch: 0 # CVE-2025-47907
+  epoch: 1
   description: Splunk OpenTelemetry Collector is a distribution of the OpenTelemetry Collector. It provides a unified way to receive, process, and export metric, trace, and log data for Splunk Observability Cloud
   copyright:
     - license: Apache-2.0
@@ -12,7 +12,6 @@ package:
 environment:
   contents:
     packages:
-      - gcc-14-default
 
 pipeline:
   - uses: git-checkout

--- a/strongswan.yaml
+++ b/strongswan.yaml
@@ -1,7 +1,7 @@
 package:
   name: strongswan
   version: "6.0.2"
-  epoch: 2
+  epoch: 3
   description: IPsec-based VPN solution focused on security and ease of use, supporting IKEv1/IKEv2 and MOBIKE
   copyright:
     # GPL-2.0-or-later WITH OpenSSL-Exception
@@ -42,7 +42,6 @@ environment:
       - ca-certificates-bundle
       - curl-dev
       - flex
-      - gcc-14-default
       - gettext-dev
       - gmp-dev
       - gperf

--- a/trafficserver-9.yaml
+++ b/trafficserver-9.yaml
@@ -1,7 +1,7 @@
 package:
   name: trafficserver-9
   version: "9.2.11"
-  epoch: 1
+  epoch: 2
   description: Apache Traffic Server™ is a fast, scalable and extensible HTTP/1.1 and HTTP/2 compliant caching proxy server.
   copyright:
     - license: Apache-2.0
@@ -14,7 +14,6 @@ environment:
       - build-base
       - busybox
       - ca-certificates-bundle
-      - gcc-14-default
       - libtool
       - linux-headers
       - openssl-dev

--- a/wasmedge.yaml
+++ b/wasmedge.yaml
@@ -1,7 +1,7 @@
 package:
   name: wasmedge
   version: "0.15.0"
-  epoch: 0
+  epoch: 1
   description: WasmEdge is a lightweight, high-performance, and extensible WebAssembly runtime for cloud native, edge, and decentralized applications.
   copyright:
     - license: Apache-2.0
@@ -20,7 +20,6 @@ environment:
       - ca-certificates-bundle
       - clang-${{vars.llvm-vers}}
       - cmake
-      - gcc-14-default
       - libLLVM-${{vars.llvm-vers}}
       - libxml2-dev
       - llvm-${{vars.llvm-vers}}

--- a/wget.yaml
+++ b/wget.yaml
@@ -1,7 +1,7 @@
 package:
   name: wget
   version: 1.25.0
-  epoch: 4
+  epoch: 5
   description: "GNU wget"
   copyright:
     - license: GPL-3.0
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       # somehow openssl-config-fipshardened-3.4.0 fails on ARMv9 with gcc-15
-      - gcc-14-default
       - openssl-dev
 
 pipeline:

--- a/yaml-cpp.yaml
+++ b/yaml-cpp.yaml
@@ -1,7 +1,7 @@
 package:
   name: yaml-cpp
   version: 0.8.0
-  epoch: 9
+  epoch: 10
   description: "yaml-cpp is a YAML parser and emitter in C++ matching the YAML 1.2 spec."
   copyright:
     - license: MIT
@@ -13,7 +13,6 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cmake-3
-      - gcc-14-default
       - wolfi-base
 
 pipeline:

--- a/yara-x.yaml
+++ b/yara-x.yaml
@@ -1,7 +1,7 @@
 package:
   name: yara-x
   version: "1.5.0"
-  epoch: 0
+  epoch: 1
   description: "A rewrite of YARA in Rust."
   copyright:
     - license: BSD-3-Clause
@@ -14,7 +14,6 @@ environment:
       - build-base
       - ca-certificates-bundle
       - cargo-c
-      - gcc-14-default
       - openssl-dev
       - perl
       - rust
@@ -54,7 +53,6 @@ test:
   environment:
     contents:
       packages:
-        - gcc-14-default
         - git
         - glibc-dev
         - go

